test/go/race_conditions_integration_test.go

```diff
@@ -37,6 +45,8 @@ func TestSelectUpdateBinlogCopy(t *testing.T) {
 			}(queries[i])
 		}
 
+		queries = annotate(queries, testcase.Ferry.Source.Marginalia)
```

Comment:
what's the purpose of annotating these queries? Are the annotations being asserted?

Reply:
The queries that are executed contain comments. This is why we need to wait for the annotated queries in this test.